modules/reindex/src/main/java/org/elasticsearch/reindex/Reindexer.java

```diff
@@ -143,7 +143,10 @@ static RestClient buildRestClient(RemoteInfo remoteInfo, ReindexSslConfig sslCon
         }).setHttpClientConfigCallback(c -> {
             // Enable basic auth if it is configured
             if (remoteInfo.getUsername() != null) {
-                UsernamePasswordCredentials creds = new UsernamePasswordCredentials(remoteInfo.getUsername(), remoteInfo.getPassword());
+                UsernamePasswordCredentials creds = new UsernamePasswordCredentials(
+                    remoteInfo.getUsername(),
+                    remoteInfo.getPassword().toString()
```

Comment:
Unfortunately, because HttpClient requires a `String` for the credentials we need to convert the `SecureString` back into a regular string here, but we can tackle that piece in the future.